server/lib/renderers/pdf.js

```diff
@@ -2,96 +2,160 @@
+    top : '5mm',
+    left : '5mm',
+    bottom : '5mm',
+    right : '5mm',
+  },
 };
 
+exports.render = renderPDF;
+exports.headers = headers;
+exports.extension = '.pdf';
+
+exports.defaultReportOptions = defaultReportOptions;
+exports.posReceiptOptions = posReceiptOptions;
+exports.reducedCardOptions = reducedCardOptions;
+
 /**
  * @function renderPDF
  *
  * @description
  * Takes in a context, template, and options before merging them and making an
- * HTML file out of the result.  The HTML file is passed to wkhtmltopdf for
+ * HTML file out of the result.  The HTML file is passed to puppeteer for
  * rendering as a PDF.
  *
  * @param {Object} context    Object of keys and values that will be made available to the handlebar template
  * @param {String} template   Path to a handlebars template
  * @returns {Promise}         Promise resolving in compiled PDF
  */
-function renderPDF(context, template, options = {}) {
-  // pdf requires absolute path to be passed to templates to be picked up by wkhtmltopdf on windows
-  context.absolutePath = path.join(process.cwd(), 'client');
-
+function renderPDF(context, template, opts = {}) {
+  const options = handleOption(opts);
   debug('received render request for PDF file. Passing to HTML renderer.');
 
   return html.render(context, template, options)
     .then((htmlStringResult) => {
       // pick options relevant to rendering PDFs
-      const pdfOptions = _.pick(options, [
-        'pageSize', 'orientation', 'pageWidth', 'pageHeight', 'marginLeft',
-        'marginRight', 'marginTop', 'marginBottom', 'footerRight',
-        'footerFontSize',
+      const _options = _.pick(options, [
+        'path', 'format', 'landscape', 'width', 'height', 'margin',
+        'headerTemplate', 'footerTemplate', 'pageRanges',
+        'printBackground', 'displayHeaderFooter',
       ]);
 
-      debug('passing rendered HTML to wkhtmltopdf.');
+      const pdfOptions = Object.assign(defaultReportOptions, _options);
+      debug('passing rendered HTML to puppeteer.');
 
-      // pass the compiled html string to the wkhtmltopdf process, this is just a wrapper for the CLI utility
-      const pdfStream = wkhtmltopdf(htmlStringResult, pdfOptions);
-
-      // this promise will only be resolved once the stream 'end' event is fired - with this implementation this will
-      // not allow the client to receive chunks of data as they are available
-      return streamToPromise(pdfStream);
+      // pass the compiled html string to puppeteer pdf generator
+      return pdfGenerator(htmlStringResult, pdfOptions);
     });
 }
+
+async function pdfGenerator(htmlString, options = {}) {
+  const browser = await browserPromise;
+  const page = await browser.newPage();
+  await page.setContent(htmlString);
+
+  // include styles
+  for (let i = 0; i < dbStyles.length; i++) {
+    const content = dbStyles[i];
+    // eslint-disable-next-line no-await-in-loop
+    await page.addStyleTag({ content : content.toString() });
```

Comment:
Since we switched from using `path:` to using `content:`, we now control the order of these scripts.  I think you can write this more efficiently by simply:
```js
await Promise.all(dbStyles);
```

Note that this will only work if you have called the `readFile` operation with 'utf8' to avoid needing to call the `.toString()` method.